lib/open_food_network/order_cycle_permissions.rb

```diff
@@ -154,49 +151,30 @@ def editable_variants_for_incoming_exchanges_from(producer)
       end
     end
 
+    def all_incoming_editable_variants
+      valid_suppliers = visible_enterprises.map do |enterprise|
+        enterprise.id if user_manages_coordinator_or(enterprise)
+      end
+
+      Spree::Variant.includes(product: :supplier).
+        joins(:product).where('spree_products.supplier_id IN (?)', valid_suppliers)
+    end
```

Comment:
We got to stop making this class longer. It's probably one of the few things we are sure won't do us any good. We should extract that into its own query objects or services or whatever you want. Besides, we already know this class is the source of several performance problems and needs to be seriously reconsidered.